Assignment4/CountingIslands.java

```diff
@@ -0,0 +1,58 @@
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+ 
+public class CountingIslands
+{
+    int CountingIslands(int noRows, int noCols, boolean[][] landOrWater) 
+	{
+		int index, indexj;
+		// Make a bool array to mark visited cells.
+	    boolean visited[][] = new boolean[noRows][noCols];
+	 
+	    // Count the islands
+	    int count = 0;
+
+	    for (index = 0; index < noRows; index++)
+	        for (indexj = 0; indexj < noCols; indexj++)
+	            if (landOrWater[index][indexj] == true && !visited[index][indexj])
+	            {
+	                DFS(index, indexj,landOrWater, visited);
+	                count = count++;
```

Comment:
This is bad in Java, absolutely horrible in C/C++, where it's undefined behaviour. You're basically saying "Get me the value of 'count', and increment it later, and take that value and assign it to 'count'." In Java, the post-increment happens before the assignment, so 'count' gets re-assigned back to its old value. In other languages, it's left undefined.

Any of the following would be ok:
count++;
++count;
count += 1;
count = count + 1;